src/request_list.js

```diff
@@ -10,6 +10,73 @@ import { getValue, setValue } from './key_value_store';
 export const STATE_PERSISTENCE_KEY = 'REQUEST_LIST_STATE';
 export const SOURCES_PERSISTENCE_KEY = 'REQUEST_LIST_SOURCES';
 
+/**
+ * @typedef {Object} RequestListOptions
+ * @property {Array.<RequestOptions|Request>} sources
```

Comment:
can drop the `.` here or use `[]`

Reply:
This one bothers me a lot, the code initializing requests is something like this:
```javascript
   for (const request in sources)
        const request = opts instanceof Request ? opts : new Request(opts);
```

So the array may contain a combination of both types. Expression `{RequestOptions[]|Request[]}` clearly says: _This is either a collection of `RequestOption`'s, or a collection of `Request`'s_.

I can't find a clear explanation of how the other form works: `Array<One|Two>` could be also expressed as the following:

```typescript
type Inner = One|Two;
type MyArray = Array<Inner>
```

This is saying to me: _This is a collection of any combination of `One`'s and `Two`'s_.

Reply:
Done.